src/awx_plugins/credentials/aws_assumerole.py

```diff
@@ -0,0 +1,182 @@
+            'type': 'string',
+            'help_text': _(
+                'The name of the key in the assumed AWS role'
+                'to fetch [AccessKeyId | SecretAccessKey | SessionToken].',
+            ),
+        },
+    ],
+    'required': [
+        'role_arn',
+    ],
+}
+
+
+def aws_assumerole_getcreds(
+        access_key: str | None,
+        secret_key: str | None,
+        role_arn: str,
+        external_id: int,
+) -> 'CredentialsTypeDef | dict[typing.Never, typing.Never]':
+    """Return the credentials for use.
+
+    :param access_key: The AWS access key ID.
+    :type access_key: str
+    :param secret_key: The AWS secret access key.
+    :type secret_key: str
+    :param role_arn: The ARN received from AWS.
+    :type role_arn: str
+    :param external_id: The external ID received from AWS.
+    :type external_id: int
+    :returns: The credentials received from AWS.
+    :rtype: dict
+    :raises ValueError: If the client response is bad.
+    """
+    connection: 'STSClient' = boto3.client(
+        service_name='sts',
+        # The following EE creds are read from the env if they are not passed:
+        aws_access_key_id=access_key,  # defaults to `None` in the lib
+        aws_secret_access_key=secret_key,  # defaults to `None` in the lib
+    )
+    try:
+        response: 'AssumeRoleResponseTypeDef' = connection.assume_role(
+            RoleArn=role_arn,
+            RoleSessionName='AAP_AWS_Role_Session1',
+            ExternalId=external_id,
+        )
+    except ClientError as client_err:
+        raise ValueError(
+            f'Got a bad client response from AWS: {client_err.message}.',
+        ) from client_err
+
+    return response.get('Credentials', {})
+
+
+def aws_assumerole_backend(
+        access_key: str | None,
+        secret_key: str | None,
+        role_arn: str,
+        external_id: int,
+        identifier: str,
+) -> dict:
+    """Contact AWS to assume a given role for the user.
+
+    :param access_key: The AWS access key ID.
+    :type access_key: str
+    :param secret_key: The AWS secret access key.
+    :type secret_key: str
+    :param role_arn: The ARN received from AWS.
+    :type role_arn: str
+    :param external_id: The external ID received from AWS.
+    :type external_id: int
+    :param identifier: The identifier to fetch from the assumed role.
+    :type identifier: str
+    :raises ValueError: If the identifier is not found.
+    :returns: The identifier fetched from the assumed role.
+    :rtype: dict
+    """
+    # Generate a unique SHA256 hash for combo of user access key and ARN
+    # This should allow two users requesting the same ARN role to have
+    # separate credentials, and should allow the same user to request
+    # multiple roles.
+    credential_key_hash = hashlib.sha256(
+        (str(access_key or '') + role_arn).encode('utf-8'),
+    )
+    credential_key = credential_key_hash.hexdigest()
+
+    credentials = _aws_cred_cache.get(credential_key, None)
```

Comment:
The tests revealed another similar bug here:
```suggestion
    credentials = _aws_cred_cache.get(credential_key, {})
```
*(https://github.com/ansible/awx-plugins/actions/runs/11265198558/job/31326677813?pr=24#step:17:155)*